pkg/nmstate/policy.go

```diff
@@ -0,0 +1,298 @@
+
+		return nil, err
+	}
+
+	return nmstatePolicy, err
+}
+
+// Exists checks whether the given NodeNetworkConfigurationPolicy exists.
+func (builder *PolicyBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof(
+		"Checking if NodeNetworkConfigurationPolicy %s exists",
+		builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Create makes a NodeNetworkConfigurationPolicy in the cluster and stores the created object in struct.
+func (builder *PolicyBuilder) Create() (*PolicyBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the NodeNetworkConfigurationPolicy %s", builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Delete removes NodeNetworkConfigurationPolicy object from a cluster.
+func (builder *PolicyBuilder) Delete() (*PolicyBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the NodeNetworkConfigurationPolicy object %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("NodeNetworkConfigurationPolicy cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete NodeNetworkConfigurationPolicy: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// Update renovates the existing NodeNetworkConfigurationPolicy object
+// with the NodeNetworkConfigurationPolicy definition in builder.
+func (builder *PolicyBuilder) Update(force bool) (*PolicyBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating the NodeNetworkConfigurationPolicy object %s",
+		builder.Definition.Name,
+	)
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+
+	if err != nil {
+		if force {
+			glog.V(100).Infof(
+				"Failed to update the NodeNetworkConfigurationPolicy object %s. "+
+					"Note: Force flag set, executed delete/create methods instead",
+				builder.Definition.Name,
+			)
+
+			builder, err := builder.Delete()
+
+			if err != nil {
+				glog.V(100).Infof(
+					"Failed to update the NodeNetworkConfigurationPolicy object %s, "+
+						"due to error in delete function",
+					builder.Definition.Name,
+				)
+
+				return nil, err
+			}
+
+			return builder.Create()
+		}
+	}
+
+	return builder, err
+}
+
+// WithInterfaceAndVFs adds SR-IOV VF configuration to the NodeNetworkConfigurationPolicy.
+func (builder *PolicyBuilder) WithInterfaceAndVFs(sriovInterface string, numberOfVF int) *PolicyBuilder {
```

Comment:
I thing numberOfVF var should have uint type

Reply:
changed uint8